### DISCUSSION TITLE
Handling "Door" alert

### DISCUSSION THREAD

**Discussion by epeleg** [winder/Universal-G-Code-Sender#1553] [Help]
Hi, 
How am I supposed to "recover" from a "Door" alert? (other then closing the door of the machine of course?)
many videos about UGS and many screen shoots in the wiki show an "Unlock" button.
but in the version I have (i.e. "Universal Gcode Platform 20201202") when the Door alert is active - the unlock button is disabled...

**Comment by breiler** [comment]
I have never used the door input and are a bit unsure on how this should work. 

I can't find anything in the GRBL documentation about the unlock command would do anything with the DOOR state. Therefore the button is inactivated and will only be activated when in a ALARM state.
https://github.com/gnea/grbl/wiki/Grbl-v1.1-Commands#x---kill-alarm-lock

**Reply by epeleg** [reply]
I am not sure myself - but there other g-code senders where the door alert gets cleared by itself when I just close the door...
maybe when detecting a door alert there should begin a loop of "?" state queries to detect when does the door alert clears and update the software state when the alert clears itself (if that happens...)?

**Reply by breiler** [reply]
That sounds neat, but for now you are required to hit play again to resume the program.

**Reply by epeleg** [reply]
I understand that.
I just want to note that this is a problematic methodology.
It caused me to crash a bit into my touch plate:
  I opened the machine, set up a the touch plate for probing closed the door and then when I tried to clear the door alert I must have accidently pressed it twice and it started running the loaded g-code instead... which I obviously did not intend to run at that stage...
